mobilecoind-dev-faucet/src/worker.rs

```diff
@@ -570,83 +590,406 @@ impl WorkerTokenState {
+        //     as change. This split tx is added to a list of at most 16 in flight
+        //     split tx's. We only take a TxOut for this purpose if it is not already
+        //     the subject of an in-flight split tx. Also, before doing anything
+        //     we check up on the in-flight split tx's.
+        let funds_are_depleted_in_top_utxos = self.maybe_send_split_txs(
+            top_utxos,
+            client,
+            monitor_id,
+            public_address,
+            target_queue_depth,
+            logger,
+        )?;
+
+        // (4) Maybe send a defragmentation tx, if funds are depleted in top utxos
+        let defragmentation_in_progress = if funds_are_depleted_in_top_utxos {
+            self.maybe_send_defragmentation_tx(
+                &non_target_value_utxos,
+                client,
+                monitor_id,
+                public_address,
+                logger,
+            )?
+        } else {
+            false
+        };
+
+        // If more split txs are needed but we can't find funds, and no defragmentation
+        // is in progress, then funds are depleted. Otherwise, funds are not
+        // depleted. Update the status.
+        if funds_are_depleted_in_top_utxos && !defragmentation_in_progress {
+            let prev_value = self.funds_depleted.swap(true, Ordering::SeqCst);
+            if !prev_value {
+                log::info!(logger, "Funds depleted on {}", self.token_id);
+            }
+        } else {
+            let prev_value = self.funds_depleted.swap(false, Ordering::SeqCst);
+            if prev_value {
+                log::info!(logger, "Funds no longer depleted on {}", self.token_id);
+            }
+        }
+
+        Ok(())
+    }
+
+    // This handles part 3 of the polling loop, where we maybe submit split or
+    // rebalancing txs.
+    //
+    // * Check on the "rebalancing" Tx process, which tries to make sure that the
+    //   top utxos are similar in value, and rebalances them if not.
+    // * Then, if we don't rebalance the top utxos, and we need more target value
+    //   utxos, make split tx's off of the top utxos in parallel (except those
+    //   already being split this way)
+    //
+    // Returns:
+    // * An error if we get a mobilecoind error
+    // * True if funds are depleted among the top utxos
+    // * False if funds are not depleted among the top utxos
+    //
+    // Assumes:
+    // top_utxos is sorted in decreasing order by value and only
+    // contains the right token id, and has the highest value MAX_OUTPUTS utxos.
+    fn maybe_send_split_txs(
+        &mut self,
+        top_utxos: &[UnspentTxOut],
+        client: &MobilecoindApiClient,
+        monitor_id: &[u8],
+        public_address: &PublicAddress,
+        target_queue_depth: usize,
+        logger: &Logger,
+    ) -> Result<bool, String> {
+        const MAX_OUTPUTS_USIZE: usize = MAX_OUTPUTS as usize;
```

Comment:
Optional/DRY: Hoist this `const` to the top-level scope. Or make `MAX_OUTPUTS` a `usize`